gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/HadoopFileSystemTestBase.java

```diff
@@ -275,6 +275,73 @@ protected void validateListFileInfo(
     }
   }
 
+  @Test
+  public void testInputStreamReadStorageStatistics() throws IOException {
+    // Write an object.
+    URI path = GoogleCloudStorageFileSystemIntegrationTest.getTempFilePath();
+    Path hadoopPath = ghfsHelper.castAsHadoopPath(path);
+    String text = "Hello World!";
+    int numBytesWritten = ghfsHelper.writeFile(hadoopPath, text, 1, /* overwrite= */ false);
+
+    try (FSDataInputStream readStream = ghfs.open(hadoopPath)) {
+      GhfsStorageStatistics StorageStats =
+          new GhfsStorageStatistics((readStream).getIOStatistics());
+      //  Check the statistics of method read().
+      readStream.read();
+      readStream.close();
+      assertThat(StorageStats.isTracked("stream_read_operations")).isTrue();
+      assertThat(StorageStats.getLong("stream_read_operations")).isEqualTo(1);
+    }
+    try (FSDataInputStream readStream = ghfs.open(hadoopPath)) {
+      GhfsStorageStatistics StorageStats =
+          new GhfsStorageStatistics((readStream).getIOStatistics());
+      // Check the statistics of method read(buf, off, len)
+      byte[] readbuffer = new byte[1];
+      readStream.read(readbuffer, 0, 1);
+      readStream.close();
+      assertThat(StorageStats.isTracked("stream_read_operations")).isTrue();
+      assertThat(StorageStats.getLong("stream_read_operations")).isEqualTo(1);
+    }
+    try (FSDataInputStream readStream = ghfs.open(hadoopPath)) {
+      GhfsStorageStatistics StorageStats =
+          new GhfsStorageStatistics((readStream).getIOStatistics());
+      // Check the statistics of method read(pos, buf, off, len)
+      byte[] readbuffer = new byte[2];
+      readStream.read(2, readbuffer, 0, 2);
+      readStream.close();
+      assertThat(StorageStats.isTracked("stream_read_operations")).isTrue();
+      assertThat(StorageStats.getLong("stream_read_operations")).isEqualTo(2);
```

Comment:
shouldn't this be a single operation ?

Reply:
Here the length is given as 2. So the no of read operation called will be 2.

Reply:
As discussed offline, The metric indicates number of calls to the API, and we are double counting in our impl.

Reply:
Updated